presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -439,7 +439,7 @@ public static NullableValue parsePartitionValue(String partitionName, String val
                 return NullableValue.of(DOUBLE, parseDouble(value));
             }
 
-            if (HIVE_STRING.equals(hiveType)) {
+            if (HIVE_STRING.equals(hiveType) || hiveType.toString().toLowerCase().startsWith("varchar")) {
```

Comment:
If you want to use `varchar` and `varchar(x)` what not check `HiveType.getTypeSignature().getBase()`?